### PR TITLE
fix ELF parse error

### PR DESCRIPTION
**Pull request by mr-tz**
Fix error encountered on various ELF files.

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

File: capa/features/extractors/elf.py [hunk 1]
```diff
@@ -458,10 +458,12 @@ def strtab(self) -> Optional[bytes]:
         for d_tag, d_val in self.dynamic_entries:
             if d_tag == DT_STRTAB:
                 strtab_addr = d_val
+                break
```
Comment (mr-tz) [inline capa/features/extractors/elf.py:461]: or is there a reason to not stop searching after finding an entry here and below?

File: capa/features/extractors/elf.py [hunk 2]
```diff
@@ -471,8 +473,10 @@ def strtab(self) -> Optional[bytes]:
 
         strtab_offset = None
         for shdr in self.section_headers:
-            if shdr.addr <= strtab_addr < shdr.addr + shdr.size:
+            # the section header address should be defined
+            if shdr.addr and shdr.addr <= strtab_addr < shdr.addr + shdr.size:
```
Comment (mr-tz) [inline capa/features/extractors/elf.py:477]: error occurred on large section with 0 start address:
```
  [26] pydata            PROGBITS         0000000000000000  0000c2fc
       00000000027b2c47  0000000000000000           0     0     1
```

File: capa/features/extractors/elf.py [hunk 3]
```diff
@@ -501,7 +505,10 @@ def needed(self) -> Iterator[str]:
             if d_tag != DT_NEEDED:
                 continue
 
-            yield read_cstr(strtab, d_val)
+            try:
+                yield read_cstr(strtab, d_val)
+            except UnicodeDecodeError as e:
+                logger.warning("failed to read DT_NEEDED entry: %s", str(e))
```
Comment (mr-tz) [inline capa/features/extractors/elf.py:511]: we should add more try/excepts around the parsing code, maybe even in/around `detect_elf_os` (best effort)

File: capa/features/extractors/elf.py (hunk 1 above)
Comment (williballenthin) [reply capa/features/extractors/elf.py:461]: as long as our tests pass, i dont have an issue with this

**Review by williballenthin — APPROVED** [review]
thank you!

File: capa/features/extractors/elf.py (hunk 1 above)
Comment (mr-tz) [reply capa/features/extractors/elf.py:461]: speaking of tests, just added one for this ELF header

**Pull request merged**